src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/begrunnelser/BegrunnelserForPeriodeContext.kt

```diff
@@ -374,12 +398,28 @@ class BegrunnelserForPeriodeContext(
             .mapNotNull { (aktør, tidslinje) ->
                 val person =
                     personopplysningGrunnlag.personer.find { it.aktør.aktivFødselsnummer() == aktør.aktivFødselsnummer() }
+                        ?: return@mapNotNull null
+
+                // VilkårResultat opphører samtidig som et annet blir oppfylt etter forskyvning
+                if (tidslinje.erTom()) {
+                    val personResultatForPerson =
+                        personResultater.find { it.aktør == aktør }
+                            ?: return@mapNotNull null
```

Comment:
Jeg er generelt ikke så fan av return@, da det blir mange exit-muligheter for løkken og det fort kan bli uoversiktlig.